pkg/kgo/record_and_fetch.go

```diff
@@ -44,7 +44,10 @@ func (a RecordAttrs) TimestampType() int8 {
 	if a.attrs&0b1000_0000 != 0 {
 		return -1
 	}
-	return int8(a.attrs & 0b0000_1000)
+	if a.attrs&0b0000_1000 == 8 {
```

Comment:
What do you think about changing this to `return (a.attrs&0b0000_1000) >> 4 == 1`?

Reply:
Hmm im fine with either but they both return a bool not an int8. and it should be >> 3 right?

Reply:
Ah right, it can be `return int8((a.attrs&0b0000_1000) >> 4)`

Reply:
smart 👍

Reply:
Nice. But still it should be >> 3. Right

`int8((8 & 0b0000_1000) >> 4) == 0` 
`int8((8 & 0b0000_1000) >> 3) == 1`